### ISSUE TITLE
Add improved custom font support

### ISSUE THREAD

**Issue by nandorojo** [nandorojo/dripsy#51]
Custom fonts in React Native are a pain. In CSS, you can do this:

```css
body {
  font-family: arial;
}
```

In React Native, you have to set the custom font on each component. This is a terrible DX.

As of Dripsy `1.3.1`, there are marked improvements for this. You can simply set a global text font by editing the `text.body` field in your theme:

```js
export default {
  text: {
    body: {
      fontFamily: 'arial'
    },
    h1: {
      fontFamily: 'arialBold'
    }
  }
}
```

Or you could do it this way:

```js
export default {
  fontFamilies: {
    body: 'arial',
    heading: 'arialBold'
  },
  text: {
    body: {
      // as of 1.3.1, text.body edits all <Text /> by default
      fontFamily: 'body'
    },
    h1: {
      fontFamily: 'heading'
    }
  }
}
```

# Open issue

But what about `fontWeight: bold`? Well, there is no concept of `@fontface` in react native. What does this mean? Rather than have a `bold` version of `arial`, if you want a `fontWeight` to apply to a custom font, you have to spell out the custom font name.

```jsx
// works...
<Text sx={{ fontFamily: 'arialBold' }}></Text>
// doesn't work.
<Text sx={{ fontFamily: 'arial', fontWeight: 'bold' }}></Text>
```

I don't think this makes any sense. What if you want to change from arial to something else? Should you really have to change the string `arialBold` everywhere in your app?

I think a nice solution could be to have a `customFonts` value in your theme that, given a font family name and fontWeight, returns a specific font.

```js
const theme = {
  customFonts: {
    arial: {
      bold: 'arialBold',
      '500': 'arialSemiBold',
    }
  }
}
```

Then, you could just do this:

```jsx
<Text sx={{ fontFamily: 'arial', fontWeight: 'bold' }}></Text>
```

...or, even better:

```jsx
const theme = {
  customFonts: {
    arial: {
      bold: 'arialBold',
      '400': 'arial',
      default: '400' // if no font family is specified, we use this, or fallback to 400
    }
  },
  fontFamilies: {
    body: 'arial',
    heading: 'arialBold'
  },
  text: {
    body: {
      fontFamily: 'body',
      fontWeight: '400'
    }
  }
}
```

And then you use it in your components like normal!

```jsx
<Text sx={{ fontWeight: 'bold' }}>This uses arialBold!</Text>
```

**Comment by cmaycumber** [comment]
Ran into this issue just the other day and ended up sticking with native fonts because of it.

I think that this would be a great addition. 

How do you imagine loading the fonts. Would you use some magic with expo-fonts behind the scene or require the user to load the fonts themselves and then some how do the matching in dripsy?

**Comment by nandorojo** [comment]
I think users will use expo fonts to handle the loading. All dripsy needs to do is match up the correct names and pass them to the component, given a font family and font weight. I'm going to try to get this working today

**Comment by nandorojo** [comment]
I'm trying this with `yarn add dripsy@custom-font` at the moment.

**Comment by nandorojo** [comment]
This is working well in my app, published to `1.4.0`.
